Zapper-fi/studio
# feat(teahouse): Add token-fetcher for Teahouse Vaults on Ethereum and Optimism

## Description


We delete teahouse.vaults.contract-position.ts on Optimism, and add teahouse.vaults.token-fetcher.ts on Ethereum and Optimism to fetch vault share's balance of address and price per share.

## Checklist


## How to test?


check vault share's balance of address and price per share.
test addresses :

Ethereum network :
0xe1b3c128c0d0a9e41ab3ff8f0984e5d5bef81677 ( share symbol : TeaETH)
-> balance about 198.77 TeaETH, price per share about : 1.00648535
0x478afa95f40bf5504cff32796c20bfd0b4e38330 (share symbol : OPTEA)
-> balance about 2651.93 OPTEA, price per share about : 1.0039168

Optimism network :
0x9ae039F9de94542f6f1b3FBA60223E6aA4f411Af ( share symbol : TeaOptions+P)
-> balance about 40893.9 TeaOptions+P, price per share about : 0.89040999
0xee1E02609a480BDC9D9651C200D90222b6691f03 ( share symbol : OPTEA+P)
-> balance about 62644.37 OPTEA+P, price per share about : 1.11617577

vaults will update price per share and balance.
